storm-core/src/clj/org/apache/storm/cluster_state/zookeeper_state_factory.clj

```diff
@@ -144,7 +145,7 @@
        (let [curator-listener (reify ConnectionStateListener
                                 (stateChanged
                                   [this client newState]
-                                  (.stateChanged listener client newState)))]
+                                  (.stateChanged listener (StormConnectionStateConverter/convert newState))))]
```

Comment:
In this line, listener is ClusterStateListener which was introduced to unify interfaces between Zookeeper and Pacemaker.

Method signature of ClusterStateLister.stateChanged() is 
`void stateChanged(org.apache.storm.cluster.ConnectionState newState)`,
 while ConnectionStateListener.stateChanged() is 
`void stateChanged(org.apache.curator.framework.CuratorFramework client, org.apache.curator.framework.state.ConnectionState newState)`

So it should remove client from parameter, and convert ConnectionState from Curator to ConnectionState from Storm.